modules/ximgproc/include/opencv2/ximgproc/structured_edge_detection.hpp

```diff
@@ -107,6 +107,26 @@ class CV_EXPORTS_W StructuredEdgeDetection : public Algorithm
     @sa Sobel, Canny
      */
     CV_WRAP virtual void detectEdges(const Mat &src, CV_OUT Mat &dst) const = 0;
+
+    /** @brief The function computes orientation from edge image.
+
+    @param src edge image.
+    @param dst orientation image.
+     */
+    CV_WRAP virtual void computeOrientation(const Mat &src, CV_OUT Mat &dst) const = 0;
+
+
+    /** @brief The function edgenms in edge image and suppress edges where edge is stronger in orthogonal direction.
+
+    @param edge_image edge image from detectEdges function.
+    @param orientation_image orientation image from computeOrientation function.
+    @param dst suppressed image (grayscale, float, in [0;1])
+    @param r radius for NMS suppression.
+    @param s radius for boundary suppression.
+    @param m multiplier for conservative suppression.
+    @param isParallel enables/disables parallel computing.
+     */
+    CV_WRAP virtual void edgesNms(const Mat &edge_image, const Mat &orientation_image, CV_OUT Mat &dst, int r = 2, int s = 0, float m = 1, bool isParallel = true) const = 0;
```

Comment:
It would be nice to replace `Mat`s to `InputArray` and `OutputArray` in public methods.